services/collectd/service.go

```diff
@@ -1,11 +1,241 @@
 package collectd
 
-import "net"
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
 
-type Service struct{}
+	"github.com/influxdb/influxdb/cluster"
+	"github.com/influxdb/influxdb/tsdb"
+	"github.com/kimor79/gollectd"
+)
 
-func NewService(c Config) *Service { return &Service{} }
+// pointsWriter is an internal interface to make testing easier.
+type pointsWriter interface {
+	WritePoints(p *cluster.WritePointsRequest) error
+}
 
-func (s *Service) Open() error    { return nil }
-func (s *Service) Close() error   { return nil }
-func (s *Service) Addr() net.Addr { return nil }
+// Service represents a UDP server which receives metrics in collectd's binary
+// protocol and stores them in InfluxDB.
+type Service struct {
+	Config       *Config
+	PointsWriter pointsWriter
+	Logger       *log.Logger
```

Comment:
No, I don't have strong feelings about this either, but I think it's mostly because we've added the `Logger` for a while as exported, and no-one is changing that. :-)

Not a blocker.